tests/utils/test_rfree.py

```diff
@@ -38,34 +43,72 @@ def test_add_rfree(data_fmodel, fraction, ccp4_convention, inplace):
-        data = rs.read_mtz(join(datadir, "9LYZ.mtz"))
-        data_rfree = rs.utils.add_rfree(data, inplace=False)
+@pytest.mark.parametrize("ccp4_convention", [False, True])
+@pytest.mark.parametrize("inplace", [False, True])
+@pytest.mark.parametrize("rfree_key", [None, "custom-rfree-key"])
+def test_copy_rfree(data_fmodel, ccp4_convention, inplace, rfree_key):
+    """
+    Test rs.utils.copy_rfree
+    """
+    data_copy = data_fmodel.copy()
 
-        # Test copy of R-free to copy of data
-        rfree = rs.utils.copy_rfree(data, data_rfree, inplace=False)
-        self.assertFalse(id(data) == id(rfree))
-        self.assertFalse("R-free-flags" in data.columns)
-        self.assertTrue("R-free-flags" in rfree.columns)
-        self.assertTrue(
-            np.array_equal(
-                rfree["R-free-flags"].values, data_rfree["R-free-flags"].values
-            )
-        )
+    # create dataset with rfree flags from which to copy
+    data_with_rfree = rs.utils.add_rfree(
+        data_fmodel, inplace=False, ccp4_convention=ccp4_convention
+    )
+
+    # handle different possible column names for rfree flags
+    if rfree_key is not None:
+        if ccp4_convention:
+            rename_dict = {"FreeR_flag": rfree_key}
+        else:
+            rename_dict = {"R-free-flags": rfree_key}
+
+        data_with_rfree.rename(columns=rename_dict, inplace=True)
+    else:
+        if ccp4_convention:
+            rfree_key = "FreeR_flag"
+        else:
+            rfree_key = "R-free-flags"
+
+    data_with_copied_rfree = rs.utils.copy_rfree(
+        data_fmodel, data_with_rfree, inplace=inplace, rfree_key=rfree_key
+    )
 
-        # Test copy of R-free inplace
-        rfree = rs.utils.copy_rfree(data, data_rfree, inplace=True)
-        self.assertTrue(id(data) == id(rfree))
-        self.assertTrue("R-free-flags" in data.columns)
-        self.assertTrue("R-free-flags" in rfree.columns)
-        self.assertTrue(
-            np.array_equal(
-                rfree["R-free-flags"].values, data_rfree["R-free-flags"].values
-            )
+    if inplace:
+        assert id(data_with_copied_rfree) == id(data_fmodel)
+        assert rfree_key in data_fmodel.columns
+        assert np.array_equal(
+            data_fmodel[rfree_key].values, data_with_rfree[rfree_key].values
+        )
+    else:
+        assert id(data_with_copied_rfree) != id(data_fmodel)
+        assert rfree_key not in data_fmodel.columns
+        assert np.array_equal(
+            data_with_copied_rfree[rfree_key].values, data_with_rfree[rfree_key].values
         )
+        assert np.all(data_fmodel == data_copy)
+
+
+def test_copy_rfree_errors(data_fmodel):
+    """
+    Test expected ValueErrors for rs.utils.copy_rfree
+    """
+    with pytest.raises(ValueError):
+        rs.utils.copy_rfree(data_fmodel, data_fmodel)
 
-        return
+    with pytest.raises(ValueError):
+        rs.utils.copy_rfree(data_fmodel, data_fmodel, rfree_key="missing key")
```

Comment:
I didn't catch this before, but this can be simplified to:
```python
@pytest.mark.parametrize("rfree_key", [None, "missing key"])
def test_copy_rfree_errors(data_fmodel, rfree_key):
    """
    Test expected ValueErrors for rs.utils.copy_rfree()
    """
    with pytest.raises(ValueError):
        rs.utils.copy_rfree(data_fmodel, data_fmodel, rfree_key=rfree_key)
 ```
 I think the above is true... let me know if I'm mistaken

Reply:
Oo good call, I think you're right. Didn't occur to me because it's a different `ValueError`, but yeah this should work.

Reply:
I'll just clarify in the docstring what the two ValueErrors are, because I think the condensed version is a little harder to parse

Reply:
In theory a slightly more rigorous test would be to confirm that the "missing key" case throws an error even when "R-free-flag" or "FreeR_flags" is present, but I think that's probably overkill.

Reply:
yep, I think your comment there is good. 

I think the test you just mentioned is a pretty good one because it tests the intended implementation (that `rfree_key` supercedes any other default column names). If we change the order of the `if` clauses in the future, that could change, so I think those sorts of tests are useful for making sure as many corner cases are covered.